src/sentry/web/frontend/twofactor.py

```diff
@@ -99,7 +99,7 @@ def validate_otp(self, otp, selected_interface, all_interfaces=None):
 
     def handle(self, request):
         user = auth.get_pending_2fa_user(request)
-        if user is None or request.user.is_authenticated():
+        if user is None or not request.user.is_authenticated():
```

Comment:
this is the big one -- it would redirect you but not clear the pending 2fa state, so now we just enforce you go through 2fa when theres a pending state

(this should never actually hit the anonymous user case)